pkg/virt-api/rest/subresource.go

```diff
@@ -1545,3 +1550,239 @@ func (app *SubresourceAPIApp) RemoveMemoryDumpVMRequestHandler(request *restful.
+	patch := fmt.Sprintf("[%s, %s]", test, update)
+
+	return patch, nil
+}
+
+func addAddInterfaceRequests(vm *v1.VirtualMachine, ifaceRequest *v1.VirtualMachineInterfaceRequest, vmCopy *v1.VirtualMachine) error {
+	canonicalIfaceName := dynamicIfaceName(ifaceRequest)
+	for _, request := range vm.Status.InterfaceRequests {
+		if err := validateAddInterfaceRequest(request, canonicalIfaceName); err != nil {
+			return err
+		}
+	}
+	vmCopy.Status.InterfaceRequests = append(vm.Status.InterfaceRequests, *ifaceRequest)
+	return nil
+}
+
+func validateAddInterfaceRequest(request v1.VirtualMachineInterfaceRequest, name string) error {
+	if addInterfaceRequestExists(request, name) {
+		return fmt.Errorf("add interface request for interface [%s] already exists", name)
+	}
+	return nil
+}
+
+func addInterfaceRequestExists(request v1.VirtualMachineInterfaceRequest, name string) bool {
+	return request.AddInterfaceOptions != nil && dynamicIfaceName(&request) == name
+}
+
+func dynamicIfaceName(plugRequest *v1.VirtualMachineInterfaceRequest) string {
+	if plugRequest.AddInterfaceOptions != nil {
+		return plugRequest.AddInterfaceOptions.InterfaceName
+	}
+	panic(fmt.Errorf("must provide the `AddInterfaceOptions.InterfaceName` attribute"))
+}
+
+func ApplyInterfaceRequestOnVMISpec(vmiSpec *v1.VirtualMachineInstanceSpec, request *v1.VirtualMachineInterfaceRequest) *v1.VirtualMachineInstanceSpec {
+	canonicalIfaceName := dynamicIfaceName(request)
+	if request.AddInterfaceOptions != nil {
+		alreadyAdded := false
+		for _, iface := range vmiSpec.Networks {
+			if iface.Name == canonicalIfaceName {
+				alreadyAdded = true
+				break
+			}
+		}
+
+		if !alreadyAdded {
+			newInterface := v1.Interface{
+				InterfaceBindingMethod: v1.InterfaceBindingMethod{Bridge: &v1.InterfaceBridge{}},
+				Name:                   canonicalIfaceName,
+			}
+			newNetwork := v1.Network{
+				Name: canonicalIfaceName,
+				NetworkSource: v1.NetworkSource{
+					Multus: &v1.MultusNetwork{
+						NetworkName: request.AddInterfaceOptions.NetworkName,
+					},
+				},
+			}
+
+			vmiSpec.Domain.Devices.Interfaces = append(vmiSpec.Domain.Devices.Interfaces, newInterface)
+			vmiSpec.Networks = append(vmiSpec.Networks, newNetwork)
+		}
+
+	}
+	return vmiSpec
+}
+
+// VMIAddInterfaceRequestHandler handles the subresource for hot plugging a network interface.
+func (app *SubresourceAPIApp) VMIAddInterfaceRequestHandler(request *restful.Request, response *restful.Response) {
+	app.addInterfaceRequestHandler(request, response, false)
+}
+
+func (app *SubresourceAPIApp) addInterfaceRequestHandler(request *restful.Request, response *restful.Response, persisted bool) {
+	name := request.PathParameter("name")
+	namespace := request.PathParameter("namespace")
+
+	if !app.clusterConfig.HotplugNetworkInterfacesEnabled() {
+		writeError(errors.NewBadRequest("Unable to Add Interface because HotplugNICs feature gate is not enabled."), response)
```

Comment:
Can you use the constant `HotplugNetworkIfacesGate` instead of this free text name?